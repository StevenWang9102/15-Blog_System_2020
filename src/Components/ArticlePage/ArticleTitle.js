import React from "react";
import { Link } from "react-router-dom";
import dateFormat from "dateformat";
import { displayLimit, offset } from "../../Functions/HttpClient";

export const ArticleTitle = props => {

  const usernameLocal =
    props.article && props.article.author && props.article.author.username;
  const imageLocal =
    props.article && props.article.author && props.article.author.image;
  const updateAtLocal =
    props.article && props.article.author && props.article.author.updatedAt;

  return (
    <Link to={`/user_profile/${usernameLocal}`}>
      <img className='author-image' src={imageLocal} alt='au' />
      <div
        className='info author'
        onClick={() => {
          props.setLoading("LOADING");

          if (props.pageName === "User Profile") {
            props.setProfileNavStatus("active", "null");
            props.loadUserProfileDetail(usernameLocal, displayLimit, offset);
          }
        }}>
        {usernameLocal}
        <span className='date'>
          {dateFormat(updateAtLocal, "ddd mmm dd yyyy")}
        </span>
      </div>
    </Link>
  );
};
