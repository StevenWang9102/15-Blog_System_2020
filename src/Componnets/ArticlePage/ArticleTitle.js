import React from "react";
import { Link } from "react-router-dom";
import dateFormat from "dateformat";

export const ArticleTitle = props => {
  return (
    <div>
      <h1>{props.currentArticleDetails.title}</h1>

      <div className='article-meta article-source'>
        <Link
          to={"/user_profile/" + props.currentArticleDetails.author.username}>
          <img
            className='author-image'
            src={props.currentArticleDetails.author.image}
            alt='au'
          />
          <div className='info author'>
            {props.currentArticleDetails &&
              props.currentArticleDetails.author &&
              props.currentArticleDetails.author.username}

            <span className='date'>
              {dateFormat(
                props.currentArticleDetails.author &&
                  props.currentArticleDetails.author.updatedAt,
                "ddd mmm dd yyyy"
              )}
            </span>
          </div>
        </Link>
      </div>
    </div>
  );
};