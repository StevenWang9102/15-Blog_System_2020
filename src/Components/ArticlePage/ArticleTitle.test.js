import React from "react";
import { ArticleTitle } from "./ArticleTitle";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ArticleTitle
        article={{}}
        pageName='test'
        setProfileNavStatus={() => {}}
        loadUserProfileDetail={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
