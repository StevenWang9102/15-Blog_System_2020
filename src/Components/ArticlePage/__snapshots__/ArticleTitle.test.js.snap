// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="#/user_profile/undefined"
  onClick={[Function]}
>
  <img
    alt="au"
    className="author-image"
  />
  <div
    className="info author"
    onClick={[Function]}
  >
    <span
      className="date"
    >
      Wed Mar 25 2020
    </span>
  </div>
</a>
`;
